File: src/main/java/com/qtech/bubbles/common/Difficulty.java
package com.qtech.bubbles.common;

/**
 * Difficulty enum, used as api for difficulty information.
 *
 * @author Quinten
 * @since 1.0.0
 */
public enum Difficulty {
    BABY(0.0625f),  // Too easy
    EASY(0.5f),
    NORMAL(1.0f),
    HARD(2.0f),
    EXPERT(4.0f),
    APOCALYPSE(8.0f),
    IMPOSSIBLE(256.0f);

    private final float plainModifier;

    Difficulty(float modifier) {
        this.plainModifier = modifier;
    }

    public float getPlainModifier() {
        return plainModifier;
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/Difficulty.kt
package com.qtech.bubbles.common

enum class Difficulty(val defaultLocal: Float) {
    BABY(0.0625f),  // Too easy
    EASY(0.5f),
    NORMAL(1.0f),
    HARD(2.0f),
    EXPERT(4.0f),
    APOCALYPSE(8.0f),
    IMPOSSIBLE(256.0f);

}